csharp/ql/src/Security Features/CWE-022/examples/TaintedPathGoodFolder.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using System.IO;
+using System.Web;
+
+public class TaintedPathHandler : IHttpHandler
+{
+    public void ProcessRequest(HttpContext ctx)
+    {
+        String filename = ctx.Request.QueryString["path"];
+        
+        string publicFolder = Path.GetFullPath("/home/" + user + "/public");
+        string filePath = Path.GetFullPath(Path.Combine(publicFolder, filename));
+
+        // GOOD: ensure that the path stays within the public folder
+        if (!filePath.StartsWith(publicFolder + Path.DirectorySeparatorChar))
```

Comment:
I don't think the query as it is catches this as "good" (I think it produces an alert).

Reply:
I think you're right. I've seen some false positives that shouldn't have been flagged.  

But that is a problem with the query, and that should probably be fixed.

Reply:
Wait, isn't this a true postive? We haven't checked that `filename` doesn't contain e.g. `../` which could remove the "public" folder part when normalized.

Reply:
> Wait, isn't this a true postive?

No.   
`Path.GetFullPath` normalizes the path so `filePath` cannot contain `../` elements. 

I constructed this example based on the examples from `java/path-injection`.  
Other languages also have similar examples, because sometimes this pattern is the correct fix.

Reply:
Right. I will open an issue for this.